### PR TITLE
Swap out History mixin in Talk

### PR DESCRIPTION
**Pull request by rogerhutchings**
Swap out the History mixin for `context.router` in Talk.

Related to #3018

### CODE REVIEW

File: app/talk/quick-subject-comment-form.cjsx [hunk 1]
```diff
@@ -19,12 +18,14 @@ PAGE_SIZE = talkConfig.discussionPageSize
 
 module.exports = React.createClass
   displayName: 'TalkQuickSubjectCommentForm'
-  mixins: [State, History]
```
Comment (rogerhutchings) [inline app/talk/quick-subject-comment-form.cjsx]: I couldn't see any of the State mixin methods used anywhere in this component...

**Comment by eatyourgreens** [comment]
Getting this error when I try to run this branch.
`Failed propType: Required prop router was not specified in TalkLatestCommentComment. Check the render method of TalkBoardDisplay.`

File: app/talk/latest-comment-link.cjsx [hunk 2]
```diff
@@ -16,13 +16,13 @@ truncate = (string = '', ending = '', length = 80) ->
...
 module.exports = React.createClass
   displayName: 'TalkLatestCommentComment'
-  mixins: [History]
 
   propTypes:
     project: React.PropTypes.object
     discussion: React.PropTypes.object
     title: React.PropTypes.bool
     preview: React.PropTypes.bool
+    router: React.PropTypes.object.isRequired
```
Comment (eatyourgreens) [inline app/talk/latest-comment-link.cjsx]: This should be in context, rather than props?

File: app/talk/latest-comment-link.cjsx [hunk 3]
```diff
@@ -16,13 +16,13 @@ truncate = (string = '', ending = '', length = 80) ->
 
 module.exports = React.createClass
   displayName: 'TalkLatestCommentComment'
```
Comment (eatyourgreens) [inline app/talk/latest-comment-link.cjsx]: Should this be `TalkLatestCommentLink`?

File: app/talk/latest-comment-link.cjsx (hunk 2 above)
Comment (rogerhutchings) [inline app/talk/latest-comment-link.cjsx]: it absolutely should

File: app/talk/latest-comment-link.cjsx (hunk 3 above)
Comment (rogerhutchings) [inline app/talk/latest-comment-link.cjsx]: you'd think :troll:

**Comment by eatyourgreens** [comment]
Cheers. There are tons of warnings about `createHref`. Should we update those calls too?

`Warning: the query argument to createHref is deprecated; use a location descriptor instead.`

**Comment by rogerhutchings** [comment]
I think we probably should - leave that with me.

**Comment by rogerhutchings** [comment]
@eatyourgreens On second thoughts, I'll fix the `createHref` calls in a separate PR - if you're happy with this, let's merge it now

**Pull request merged**